rclcpp/src/rclcpp/qos.cpp

```diff
@@ -16,6 +16,15 @@
 
 #include <rmw/types.h>
 
+namespace
+{
+/// Check if two rmw_time_t have the same values.
+bool equal(const rmw_time_t & left, const rmw_time_t & right)
```

Comment:
Why not make this `bool operator==()`?

Reply:
Can't define an operator in an anonymous namespace, but wanted to make sure not to export the symbol

Reply:
You mean the issue is callers of this `<anonymous>::operator==()` function couldn't find it, right?

Is `operator==(const QoS & left, const QoS & right)` the only function that calls this? Maybe try nesting this anonymous namespace under the `rclcpp` namespace, where `operator==(const QoS & left, const QoS & right)` resides.

Reply:
Yes, that operator is the only place this is used. Didn't realize I could nest the unnamed namespace, that does make it resolve here. Pushing a new commit